src/views/documents/SearchStore.ts

```diff
@@ -11,25 +11,29 @@ export interface SearchItem {
   journalId: string;
 }
 
+interface SearchQuery {
+  journals: string[];
+  titles?: string[];
+  before?: string;
+  texts?: string[];
+  limit?: number;
+}
+
 export class SearchV2Store {
   @observable docs: SearchItem[] = [];
   @observable loading = true;
   @observable error: string | null = null;
   private journals: JournalsStore;
   private tagSeachStore: TagSearchStore;
-  private setTokensUrl: any; // todo: This is react-router-dom's setUrl; type it
+  setTokensUrl: any; // todo: This is react-router-dom's setUrl; type it
 
   @observable private _tokens: IObservableArray<SearchToken> = observable([]);
 
-  constructor(private client: IClient, journals: JournalsStore, setTokensUrl: any) {
+  constructor(private client: IClient, journals: JournalsStore, setTokensUrl: any, tokens: string[]) {
     this.journals = journals;
     this.tagSeachStore = new TagSearchStore(this);
     this.setTokensUrl = setTokensUrl;
-
-    // Re-run the search query anytime the tokens change.
-    reaction(() => this._tokens.slice(), this.search, {
```

Comment:
I invariably end up replacing reactions with imperative code. Reaction always feels more elegant, but `@action` and `@computed` and `@observable` alone are usually easier to reason about.